### ISSUE TITLE
Analogue Pocket OS update December 2023

### ISSUE THREAD

**Issue by ThePrinceOfTroy** [spiritualized1997/openFPGA-GB-GBC#48]
When will support for openfpga display filters be added?

**Comment by DunhamGitHub** [comment]
Filters were added December 25 2023 via 2.0 Firmware.

**Comment by SaulGoodman1337** [comment]
> Filters were added December 25 2023 via 2.0 Firmware.

A little tip for your future life:
If you have no idea what you're talking about, you should keep quiet.

**Comment by ThePrinceOfTroy** [comment]
> Filters were added December 25 2023 via 2.0 Firmware.

Right, but does the core itself need an update? I've updated my Pocket and I still have no ability to use the filters. Maybe I'm doing something wrong

**Comment by DunhamGitHub** [comment]
> > Filters were added December 25 2023 via 2.0 Firmware.
> 
> A little tip for your future life: If you have no idea what you're talking about, you should keep quiet.

Right back at you

**Comment by DunhamGitHub** [comment]
> > Filters were added December 25 2023 via 2.0 Firmware.
> 
> Right, but does the core itself need an update? I've updated my Pocket and I still have no ability to use the filters. Maybe I'm doing something wrong

Play any .gbc / .gb game, press the Analogue-button. Does it show "display mode"? In any case ...post a photo why don't you

**Comment by SaulGoodman1337** [comment]
> > Filters were added December 25 2023 via 2.0 Firmware.
> 
> Right, but does the core itself need an update? I've updated my Pocket and I still have no ability to use the filters. Maybe I'm doing something wrong

just ignore the guy... He has absolutely no idea what he's talking about. You're right, the core needs to be updated.

I just had a look and 2 hours ago a project popped up on Github that does exactly what we want.

https://github.com/budude2/openfgpa-GBC

I just tested it and the display modes work ! The core also offers RTC support.

Give it a try, if you have any questions feel free to contact me.


![PXL_20240103_020753193](https://github.com/spiritualized1997/openFPGA-GB-GBC/assets/4953452/cf754f87-a5b3-4c67-8e26-b6a5f18ba69f)
![PXL_20240103_020820981](https://github.com/spiritualized1997/openFPGA-GB-GBC/assets/4953452/a40f7072-fb97-468e-aa0f-8e6f92ff7ea3)

Edit:

One problem at the moment, no save states. But it is in progress

**Comment by DunhamGitHub** [comment]
I don't see what the OP means if not this...

![IMG_3611](https://github.com/spiritualized1997/openFPGA-GB-GBC/assets/6884752/57e6d1fa-3056-4cf1-93d6-07f6b94b410d)


@SaulGoodman1337 please do tell how I am confused about the topic!

**Comment by JoneWang** [comment]
@SaulGoodman1337  You can refer to this list and add filters to the core yourself:
https://www.analogue.co/developer/docs/core-definition-files/video-json#display-mode-id-table

Here's an example of a GBC core addition to spiritualized1997:
**Pocket:\Cores\Spiritualized.GBC\video.json**
```
{
    "video": {
        "magic": "APF_VER_1",
        "scaler_modes": [
            {
                "width": 160,
                "height": 144,
                "aspect_w": 160,
                "aspect_h": 144,
                "rotation": 0,
                "mirror": 0
            }
        ],
        "display_modes": [
            {
                "id": "0x10"
            },
            {
                "id": "0x20"
            },
            {
                "id": "0x30"
            },
            {
                "id": "0x31"
            },
            {
                "id": "0x32"
            },
            {
                "id": "0x40"
            },
            {
                "id": "0xE0"
            }
        ]
    }
}
```

**Comment by SaulGoodman1337** [comment]
Tell me, am I in a madhouse here? 
OP is talking about Modes:
- Original GB DMG
- Original GBP
- Original GBP Light 

which are currently not possible because the core needs an Update. OP is absolutely right here and you two have absolutely no idea what you are talking about. 
But I don't care now, live in your world.

![PXL_20240103_122957439 MP](https://github.com/spiritualized1997/openFPGA-GB-GBC/assets/4953452/ef20f644-3bf8-4b48-bd76-af7f3d73b98c)
![PXL_20240103_123003075](https://github.com/spiritualized1997/openFPGA-GB-GBC/assets/4953452/3762a4a9-12c8-4f98-aae2-dbda791c091b)
![PXL_20240103_123007547](https://github.com/spiritualized1997/openFPGA-GB-GBC/assets/4953452/4f427950-8644-4ea0-900a-6beabd3e3096)
![PXL_20240103_123450367](https://github.com/spiritualized1997/openFPGA-GB-GBC/assets/4953452/b79f2aec-79d9-41d1-9b9a-7ced6a36b64f)


@JoneWang 
It's even in your link....
For display modes:
- 0x21 Original GB DMG
- 0x22 Original GBP
- 0x23 Original GBP Light

**Requires core response, which is not possible with the current Spiritualized Core**.
But the main thing is to post a link that you don't understand yourself. Bravo.

**Comment by DunhamGitHub** [comment]
> Tell me, am I in a madhouse here? OP is talking about Modes:
> 
> * Original GB DMG
> * Original GBP
> * Original GBP Light
> 

(sure those modes are not supported by this! core, fair enough)

Perhaps, you are (over)-assuming that the OP is actually talking about just those individual modes and not in a more general sense
...Perhaps he just never used one of the Auto-Updaters (like Matt-Panella) that adds the configs to the .json files automatically ...as simple as that🫣

**Comment by SaulGoodman1337** [comment]
> > Tell me, am I in a madhouse here? OP is talking about Modes:
> > 
> > * Original GB DMG
> > * Original GBP
> > * Original GBP Light
> 
> (sure those modes are not supported by this! core, fair enough)
> 
> Perhaps, you are (over)-assuming that the OP is actually talking about just those individual modes and not in a more general sense ...Perhaps he just never used one of the Auto-Updaters (like Matt-Panella) that adds the configs to the .json files automatically ...as simple as that🫣

Please stop suggesting that you had any idea what this was actually about. You don't.
I'm out at this point. 

@ThePrinceOfTroy 
Have fun with the display modes

**Comment by DunhamGitHub** [comment]
> 
> Please stop suggesting that you had any idea what this was actually about. You don't. I'm out at this point.
> 

Nor did you, obviously🙊 (*But yes I am aware for those Modes the Core would need an update, but the OP did not ask that)
